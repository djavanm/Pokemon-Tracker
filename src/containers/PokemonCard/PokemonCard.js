import React from 'react';
import { connect } from 'react-redux';

export const PokemonCard = ({ currentPokemon }) => {
  const { name, id, types, height, weight, image, gen } = currentPokemon;
  let typeInfo = types ? types.map((type, index) => <li key={index}>{type}</li>) : null;
  return (
      <article>
      <img src={image} alt={`name's card`} data-id={id} />
      <p>{name}</p>
      <p>Type{types.length > 1 && <span>s</span>}</p>
      <ul>
      {typeInfo}
      </ul>
      <p>Height: {height}</p>
      <p>Weight: {weight}</p>
      <p>Gen: {gen}</p>
    </article>
  )
}

export const mapStateToProps = state => ({
  currentPokemon: state.currentPokemon
});

export default connect(mapStateToProps)(PokemonCard);
