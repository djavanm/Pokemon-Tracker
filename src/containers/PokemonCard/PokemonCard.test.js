import React, { Component } from 'react';
import { shallow } from 'enzyme';
import { PokemonCard, mapStateToProps } from './PokemonCard';


describe('PokemonCard', () => {
  let wrapper;
  const mockPokemon = {
    name: 'Bulbasaur',
    description: 'Bulbasaur is cool.',
    gen: 'Generation 1',
    gif: 'https://projectpokemon.org/images/normal-sprite/bulbasaur.gif',
    height: 7,
    id: 1,
    image: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png',
    sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png',
    types: ['Grass', 'Poison']
  };

  const mockCaughtPokemon = [mockPokemon];
  const mockTeamPokemon = [mockPokemon];
  const mockCurrentPokemon =  mockPokemon;
  const mockState = {
    currentPokemon: mockPokemon,
    caughtPokemon: mockCaughtPokemon,
    teamPokemon: mockTeamPokemon
  };

  beforeEach(() => {
    wrapper = shallow(<PokemonCard
      currentPokemon={mockPokemon}
      />);
  });

  it('should match the snapshot given the correct information', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('mapStateToProps should grab the props it needs', () => {
    const expected = {
      currentPokemon: mockPokemon
    }
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });

});
