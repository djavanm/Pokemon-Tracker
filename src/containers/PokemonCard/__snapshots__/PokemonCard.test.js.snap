// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard should match the snapshot given the correct information 1`] = `
<article
  className="card-container"
>
  <img
    alt="name's card"
    className="poke-img"
    data-id={1}
    src="https://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png"
  />
  <div
    className="poke-info-container"
  >
    <div
      className="poke-info"
    >
      <p
        className="pokemon-banner"
      >
        Current Pokemon:
      </p>
      <p
        className="pokemon-name"
      >
         
        Bulbasaur
         
      </p>
      <p
        className="pokemon-description"
      >
        Bulbasaur is cool.
      </p>
    </div>
    <div
      className="poke-stats"
    >
      <div
        className="box"
      >
        <p>
          Type
          <span
            className="span-type"
          >
            s
          </span>
          :
        </p>
        <ul
          className="poke-info-list"
        >
          <li
            className="grass"
            key="0"
          >
            Grass
          </li>
          <li
            className="poison"
            key="1"
          >
            Poison
          </li>
        </ul>
      </div>
      <div
        className="box measurements"
      >
        <p>
          Height: 
        </p>
        <p>
          7
          ft
        </p>
        <p>
          Weight:
        </p>
        <p>
           lbs
        </p>
      </div>
      <div
        className="box"
      >
        <p
          className="pokedex-number"
        >
          #
          1
        </p>
        <p
          className="generation"
        >
          Generation 1
        </p>
      </div>
    </div>
  </div>
</article>
`;
